docs/recipes/patterns/add-dependency-injection.md

```diff
@@ -0,0 +1,56 @@
+> This recipe is not a detailed discussion of the pros and cons of Dependency Injection (DI) compared to other patterns. It simply illustrates how to use it within a SAFE Stack application!
+
+1. Create a class that you wish to inject with a dependency (in this example, we use the built-in `IConfiguration` type that is included in ASP .NET):
+
+    ```fsharp
+    open Microsoft.Extensions.Configuration
+
+    type DatabaseRepository(config:IConfiguration) =
+        member _.SaveDataToDb (text:string) =
+            let connectionString = config["SqlConnectionString"]
+            // Connect to SQL using the above connection string etc.
+            Ok 1
+    ```
+
+    > Instead of functions or modules, DI in .NET and F# only works with classes.
+
+2. Register your type with ASP .NET, typically in the start up / bootstrapper.
```

Comment:
In other recipes we assume they work from a vanilla SAFE template, so no need for typically